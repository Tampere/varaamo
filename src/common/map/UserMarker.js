import L from 'leaflet';
import shadowUrl from 'leaflet/dist/images/marker-shadow.png';
import PropTypes from 'prop-types';
import React from 'react';
import { Marker } from 'react-leaflet';

import iconRetinaUrl from '../../../app/assets/map/usermarker-icon-2x.png';
import iconUrl from '../../../app/assets/map/usermarker-icon.png';

const UserMarker = ({ position }) => (
  <Marker
    icon={L.icon({
      iconAnchor: [12, 41],
      iconRetinaUrl,
      iconSize: [25, 41],
      iconUrl,
      popupAnchor: [1, -34],
      shadowSize: [41, 41],
      shadowUrl,
      tooltipAnchor: [16, -28],
    })}
    position={position}
  />
);

UserMarker.propTypes = {
  position: PropTypes.arrayOf(Number),
};

export default UserMarker;
