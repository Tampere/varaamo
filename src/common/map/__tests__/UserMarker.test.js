import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import UserMarker from '../UserMarker';

describe('UserMarker', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<UserMarker position={[60.20843, 24.975634]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
