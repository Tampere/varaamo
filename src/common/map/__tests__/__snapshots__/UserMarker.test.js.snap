// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMarker renders correctly 1`] = `
<ForwardRef(Leaflet(Marker))
  icon={
    NewClass {
      "_initHooksCalled": true,
      "options": Object {
        "iconAnchor": Array [
          12,
          41,
        ],
        "iconRetinaUrl": "test-file-stub",
        "iconSize": Array [
          25,
          41,
        ],
        "iconUrl": "test-file-stub",
        "popupAnchor": Array [
          1,
          -34,
        ],
        "shadowSize": Array [
          41,
          41,
        ],
        "shadowUrl": "test-file-stub",
        "tooltipAnchor": Array [
          16,
          -28,
        ],
      },
    }
  }
  position={
    Array [
      60.20843,
      24.975634,
    ]
  }
/>
`;
